### PR TITLE
Add more metrics

### PR DESCRIPTION
**Pull request by dan437**
## Explanation
This PR adds some more metrics for MetaMask:
- `App Installed` (only if a user opted into metrics)
- `signup_date` tracking via traits
- `Support Link Clicked` for all links that use the `SUPPORT_REQUEST_LINK` constant
- `New Wallet Created` event after a user clicks on `Remind me later` (skips seed backup)
- `App Uninstalled` via a URL that needs to be implemented first, currently this code is disabled
- Track `Support Link Clicked` events also for the `SUPPORT_LINK` constant

### `App Installed` and `New Wallet Created` events
- Remove the extension
- In `scripts/background.js` remove this condition: `!(process.env.METAMASK_DEBUG || process.env.IN_TEST)` from the `browser.runtime.onInstalled` listener
- Run `yarn start`
- Install the extension
- Check the background.html dev tools, specifically the Network tab and search for `batch`, which shows our metrics requests
- After you enable metrics, you will see that the `App Installed` event is tracked
- Create a new wallet and look for the `New Wallet Created` event when you click on `Remind me later` to skip seed backup

### `Support Link Clicked`
- Open the extension, click on your profile (top right corner)
- Click on `Support`, which opens a new tab
image
- Verify that the `Support Link Clicked` event was tracked if you opted in to metrics

### CODE REVIEW

File: app/scripts/controllers/metametrics.js [hunk 1]
```diff
@@ -315,6 +319,24 @@ export default class MetaMetricsController {
     this._identify(allValidTraits);
   }
 
+  // It sets an uninstall URL ("Sorry to see you go!" page),
+  // which is opened if a user uninstalls the extension.
+  updateExtensionUninstallUrl(participateInMetaMetrics, metaMetricsId) {
+    // TODO: Change it to the right URL once it's available.
+    const extensionUninstallUrl =
+      'https://swap.metaswap.codefi.network/tracking/uninstall';
```
Comment (brad-decker) [inline app/scripts/controllers/metametrics.js:327]: lets make this url a constant at the top of this file.

File: ui/store/actions.js [hunk 2]
```diff
@@ -3802,3 +3802,7 @@ export function requestAddNetworkApproval(customRpc, originIsMetaMask) {
     }
   };
 }
+
+export async function addEventBeforeMetricsOptIn(event) {
+  await promisifiedBackground.addEventBeforeMetricsOptIn(event);
+}
```
Comment (brad-decker) [inline ui/store/actions.js:3808]: Remove this. We likely will never have a need for this from the UI. You likely meant to as there isn't a corresponding change in metamask-controllers.

**Review by brad-decker — CHANGES_REQUESTED** [review]

**Comment by brad-decker** [comment]
@worldlyjohn Could you review the schema/format of these events?

File: app/scripts/background.js [hunk 3]
```diff
@@ -751,12 +754,31 @@ async function openPopup() {
...
 }
 
+// It adds the "App Installed" event into a queue of events, which will be tracked only after a user opts into metrics.
+const addAppInstalledEvent = () => {
+  if (controller) {
+    controller.metaMetricsController.addEventBeforeMetricsOptIn({
+      category: EVENT.CATEGORIES.APP,
+      event: EVENT_NAMES.APP_INSTALLED,
+      properties: {
+        signup_date: new Date().toISOString().split('T')[0], // yyyy-mm-dd
```
Comment (worldlyjohn) [inline app/scripts/background.js:764]: @dan437 and i already discussed we wanted to move this property, `signup_date`, out of this event and into a user trait. properties list should be `{}` here

**Comment by worldlyjohn** [comment]
aside from moving `signup_date` from event property to user trait... everything else looks good from a schema POV!

File: app/scripts/controllers/metametrics.js (hunk 1 above)
Comment (worldlyjohn) [reply app/scripts/controllers/metametrics.js:327]: i'm working with website team to define endpoint as `https://metamask.io/uninstalled`. 
Can track [status of ticket](https://github.com/ConsenSys/dx-team/issues/1339) for Website team, but not yet implemented (as of this comment)

File: app/scripts/controllers/metametrics.js [hunk 4]
```diff
@@ -96,12 +100,16 @@ export default class MetaMetricsController {
...
+    this.extension = extension;
+    this.environment = environment;
 
     const abandonedFragments = omitBy(initState?.fragments, 'persist');
 
     this.store = new ObservableStore({
       participateInMetaMetrics: null,
       metaMetricsId: null,
+      eventsBeforeMetricsOptIn: [],
+      traits: {},
```
Comment (brad-decker) [inline app/scripts/controllers/metametrics.js:112]: [REQ] The MetaMetricsControllerState jsdoc type needs updated in this file to match these new values.

**Review by brad-decker — CHANGES_REQUESTED** [review]
I like the addition of the traits object in metametrics state for transient traits not derived from other state keys. One minor request is just to update the docs for the state type. Thanks @dan437

**Review by brad-decker — DISMISSED** [review]

**Comment by brad-decker** [comment]
re-triggered tests cause the unit test failure is a flaky test

**Review by brad-decker — DISMISSED** [review]
LGTM

**Review by brad-decker — DISMISSED** [review]

**Review by digiwand — COMMENTED** [review]
Logic looks good! No blockers. I left a few suggestions about naming and reordering props

File: ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js [hunk 5]
```diff
@@ -76,6 +79,15 @@ export default class RevealSeedPhrase extends PureComponent {
       },
     });
 
+    this.context.trackEvent({
+      category: EVENT.CATEGORIES.ONBOARDING,
+      event: EVENT_NAMES.NEW_WALLET_CREATED,
+      properties: {
+        action: 'Onboarding Complete',
+        legacy_event: true,
```
[the remark this answers is not in the document]
Comment (dan437) [reply ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js:87]: It should be only used for legacy events and not for new ones. We already have `legacy_event: true` for the `New Wallet Created` event in a different place and that's why I have it here as well.

File: ui/contexts/metametrics.js [hunk 6]
```diff
@@ -57,11 +58,26 @@ export function MetaMetricsProvider({ children }) {
   const location = useLocation();
   const context = useSegmentContext();
 
+  // Sometimes we want to track context fields inside the "properties" object.
+  const addContextFieldsIntoProperties = (payload, options) => {
```
[the remark this answers is not in the document]
Comment (dan437) [reply ui/contexts/metametrics.js:62]: Always happy to make code easier to understand! I've renamed context fields into context props in other places as well.

**Review by brad-decker — DISMISSED** [review]

File: ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (brad-decker) [reply ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js:87]: @worldlyjohn do we want to keep this event, moving it from saying 'action' and give it a fully formed schema or do we want to keep it as legacy_event: true to mark it for future refactoring?

File: ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js [hunk 7]
```diff
@@ -76,6 +79,15 @@ export default class RevealSeedPhrase extends PureComponent {
...
+      },
+    });
+
     await Promise.all([setCompletedOnboarding(), setSeedPhraseBackedUp(false)]);
```
Comment (digiwand) [inline ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js:91]: I don't think this is a blocker, but we could move the new `EVENT_NAMES.NEW_WALLET_CREATED`, action: `Onboarding Complete`, underneath this await to better ensure that the onboarding is complete

**Review by digiwand — DISMISSED** [review]
Thanks for the updates! 🙏

File: ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (worldlyjohn) [reply ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js:89]: @brad-decker  Yep I noticed this and think it's better to conform to the current standard.  Someone can then change all the `New Wallet Created` events (including this one) in one swoop.

**Review by digiwand — APPROVED** [review]
🙌

**Review by brad-decker — APPROVED** [review]

**Pull request merged**